api/query_builder.go

```diff
@@ -268,8 +292,28 @@ var PullRequestFields = append(IssueFields,
 	"statusCheckRollup",
 )
 
+type issueGraphQLOpts struct {
+	useCheckRunAndStatusContextCounts bool
+}
+
+type IssueGraphQLOptFn func(*issueGraphQLOpts)
+
+func WithUseCheckRunAndStatusContextCounts() IssueGraphQLOptFn {
+	return func(opts *issueGraphQLOpts) {
+		opts.useCheckRunAndStatusContextCounts = true
+	}
+}
+
 // IssueGraphQL constructs a GraphQL query fragment for a set of issue fields.
-func IssueGraphQL(fields []string) string {
+func IssueGraphQL(fields []string, opts ...IssueGraphQLOptFn) string {
```

Comment:
Not really bothered about using functional option pattern here but it was an easy way to get the option passed down without updating all the call sites with a new arg.

Reply:
That sounds like a good idea. I also like the fact that it removes the nested conditional under `statusCheckRollup`. One thing that seems to change is that the name of the `field` is no longer _actually_ the name the of the GQL field. I think that's ok as long as we document it properly, such as with the `pseudo-field` branches. What do you think?

This is somewhere that having real enum/discriminated union support would be really nice to associate the boolean with only one variant, but alas in Go!

Reply:
`statusCheckSummary` would qualify as a `pseudo-field`. We just use that term to mean a field where the name does not directly translate to the GraphQL object/field it retrieves. It is totally fine to add them, when possible we do try to keep the names the same as the API though so the fields can be looked up in the API docs easily. We actually do not do a good job at documenting the fields that are available because we haven't found a good place to display them. We have an issue somewhere about better documentation for them. Right now the user has to run the given command with the `--json` flag to see the list of available flags, and they do not have any description next to them.